openmedia/gui/frames.py

```diff
@@ -1,189 +1,179 @@
+        mixer.add_observer(self)
+        self._create_control_widgets()
+        self._create_playlist()
+        self.progress = Gtk.Scale()
+        self.progress.set_range(0, mixer.get_song_duration())
+        self.progress.connect("button-press-event", self._start_skip)
+        self.progress.connect("button-release-event", self._end_skip)
+        self.progress.set_vexpand(False)
+
+        self.leftBox = Gtk.VBox()
+        self.leftBox.pack_start(self.buttonsBox, True, False, 0)
+        self.leftBox.pack_start(self.progress, False, False, 0)
+        self.leftBox.set_vexpand(False)
+
+        self.mainHBox = Gtk.VBox()
+        self.mainHBox.set_spacing(5)
+        self.mainHBox.pack_start(self.playlistBox, False, False, 0)
+        self.mainHBox.pack_start(self.leftBox, False, True, 0)
+        self.mainHBox.set_vexpand(False)
+
+        self.add(self.mainHBox)
+        self.set_vexpand(False)
+        self.set_default_size(-1, 120)
+
+    def _create_control_widgets(self):
+        self.play_button = Gtk.Button(label=u'▶')
+        self.play_button.connect("clicked", self._play_pause)
+        self.play_button.set_vexpand(False)
+
+        self.stop_button = Gtk.Button(label=u'■')
+        self.stop_button.connect("clicked", self._stop)
+        self.stop_button.set_vexpand(False)
+
+        self.play_next_button = Gtk.Button(label=">>")
+        self.play_next_button.connect("clicked", self._play_next)
+        self.play_next_button.set_vexpand(False)
+
+        self.volume_button = Gtk.VolumeButton()
+        self.volume_button.set_value(0.5)
+        self.volume_button.connect("value-changed", self._volume)
+        self.volume_button.x_align = 1.0
+        self.volume_button.set_vexpand(False)
+
+        self.playlist_button = Gtk.ToggleButton(label="=")
+        self.playlist_button.connect("clicked", self._toggle_playlist)
+        self.playlist_button.set_vexpand(False)
+
+        self.buttonsBox = Gtk.HBox(False)
+        self.buttonsBox.set_spacing(5)
+        self.buttonsBox.pack_start(self.play_button, False, False, 0)
+        self.buttonsBox.pack_start(self.stop_button, False, False, 0)
+        self.buttonsBox.pack_start(self.play_next_button, False, False, 0)
+        self.volumeBox = Gtk.HBox()
+        self.volumeBox.pack_end(self.volume_button, False, True, 0)
+        self.buttonsBox.pack_start(self.playlist_button, False, False, 0)
+        self.buttonsBox.pack_end(self.volumeBox, True, True, 0)
+        self.buttonsBox.set_vexpand(False)
 
-    def _play_pause(self, button):
-        self._update_play_icon(button)
+    def _create_playlist(self):
+        self.model = Gio.ListStore.new(ModelItem)
+        self.playlist = Gtk.ListBox()
+        self.playlist.bind_model(self.model, self._create_list_item, None)
+        self.playlist.connect("row_activated", self._play_item)
+        for track in mixer.track_list:
+            item = ModelItem(track.metadata.track_name, track.duration)
```

Comment:
We could create a few properties which would actually use the `metadata` object that is created.